### DISCUSSION TITLE
Datadog APM on Google Cloud Run

### DISCUSSION THREAD

**Discussion by antikhv** [serversideup/docker-php#292] [Q&A]
Hi folks 👋 

We are using 8.1-fpm-nginx of the Docker image and tried to instrument our service following the documentation from [Datadog](https://docs.datadoghq.com/serverless/google_cloud_run/?code-lang=php). Because of the s6 overlay, we added the serverless-init as an entrypoint script but received this error `s6-overlay-suexec: fatal: can only run as pid 1` . Has anyone successfully instrumented their service and forwarded traces to Datadog using the documented setup? Would love to see some insights, thank you!

**Comment by jaydrogers** [comment]
Sorry for the delayed response... My wife and I had our first child last month 🤪

Have you considered using a different variation? NGINX + FPM requires S6 Overlay, but S6 Overlay is very picky: https://serversideup.net/open-source/docker-php/docs/guide/using-s6-overlay#this-disadvantage-of-s6-overlay

I am hoping that FrankenPHP or NGINX Unit will be the modern replacement for running Docker + PHP

**Comment by papoms** [comment]
Using serverless-init as an entrypoint script did not work for us either.
We install DD Agent[1] in the Dockerfile and start the DD Agent in a S6 user script.
```
#!/command/with-contenv bash

# Start the Datadog agent
service datadog-agent start
```


-- 
[1] To be more specific, the actual install is also a S6 Oneshot, because install_script_agent7.sh expects DD_API_KEY to be present. But we only have DD_API_KEY once the actual container runs, not during build.

**Reply by jaydrogers** [reply]
I havent used `datadog-agent` before, but make sure that `www-data` has permissions to start that service. Otherwise you'll need to escalate privileges to `USER root` before attempting to run that.
